### PR TITLE
docs: add multi-tenancy docs

### PR DESCRIPTION
**Pull request by carvantes**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: USING_MULTI_TENANCY.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Enabling multi-tenancy
+
+Use the `enableMultiTenancy` option when deploying the stack: 
+
+```bash
+serverless deploy --enableMultiTenancy true
+```
+
+**Note:** Updating an existing (single-tenant) stack to enable multi-tenancy is a breaking change. Multi-tenant
```
Comment (Bingjiling) [inline USING_MULTI_TENANCY.md:17]: Is it worth mentioning once multi-tenancy is turned on, turning back to single tenant mold is also a breaking change?

File: USING_MULTI_TENANCY.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+**Note:** Updating an existing (single-tenant) stack to enable multi-tenancy is a breaking change. Multi-tenant 
+deployments use a different data partitioning strategy that renders the old, single-tenant, data inaccessible. 
+If you wish to switch from single-tenant to a multi-tenant model, it is recommended to create a new multi-tenant stack 
+and then migrate the data from the old stack. 
+
+## Tenant identifiers
+
+Tenants are identified by a tenant Id in the auth token. There are 2 ways to include a tenant Id in the auth token:
+
```
Comment (Bingjiling) [inline USING_MULTI_TENANCY.md:28]: Mention syntax the tenantId should follow?

**Review by Bingjiling — APPROVED** [review]
Looks good! Some nit picks:

File: USING_MULTI_TENANCY.md (hunk 2 above)
Comment (carvantes) [reply USING_MULTI_TENANCY.md:28]: Done

File: USING_MULTI_TENANCY.md (hunk 1 above)
Comment (carvantes) [reply USING_MULTI_TENANCY.md:17]: I added a small note about it.

**Pull request merged**